slave_log_event.h

```diff
@@ -87,6 +87,29 @@ enum Log_event_type
 
   XA_PREPARE_LOG_EVENT= 38,
 
+  /*
+   * Extension of UPDATE_ROWS_EVENT, allowing partial values according
+   * to binlog_row_value_options.
+   */
+  PARTIAL_UPDATE_ROWS_EVENT = 39,
+
+  /*
+   * Add new events here - right above this comment!
+   * Existing events (except ENUM_END_EVENT) should never change their numbers
+   */
+
+  /* New MySQL events are to be added right above this comment */
+  MYSQL_END_EVENT,
+
+  /* Add new Percona Server events here - its ids should go downwards
+   * starting from MARIA_EVENTS_BEGIN, i.e. 159, 158 ..
+   * till MYSQL_END_EVENT
+   */
+
+  START_5_7_ENCRYPTION_EVENT = 159,
+
+  MARIA_EVENTS_BEGIN = 160,
+
```

Comment:
`libslave` has a unit test (`test/unit_test.cpp`) that covers different types of events. It would be good to add new testcases for a new event types.